src/agent.py

```diff
@@ -67,6 +67,8 @@ def _setup_llm_provider(self):
         # Load Twitter username for self-reply detection
         load_dotenv()
         self.username = os.getenv('TWITTER_USERNAME', '').lower()
+        if not self.username:
+                raise KeyError("Twitter username is required")
```

Comment:
Can we use a different error type here (maybe ValueError)? I think KeyError aer used is used for accessing a non existing keys in a a dictionary https://www.turing.com/kb/python-keyerror-exception.